File: js/mgtFaeries.js
var nm1 = ["","","","","c","f","h","l","n","r","s","w","z"];
var nm2 = ["a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","y","oo","ia","ea","ee"];
var nm3 = ["b","d","l","ll","m","mm","n","nn","r","s","v","b","d","dw","l","ll","lw","lr","lm","ln","m","mr","mm","n","nm","nr","nv","r","rl","rn","rm","sh","sn","sr","vl","vr"];
var nm4 = ["c","d","l","n","r","s","th","z"];
var nm5 = ["","","","","","","","","","","","","","","","h","l","n","ph","s","th"];

var nm6 = ["acorn","alder","alpen","amber","autumn","barely","beech","birch","briar","bright","cedar","cherry","cinder","cloud","crystal","dawn","dew","dream","dusk","elm","ember","feather","fern","fog","forest","free","frost","gentle","grand","grass","great","green","haven","hazel","heart","holly","humble","keen","kind","leaf","light","lone","maple","marble","marsh","mellow","mist","moon","moss","nettle","night","oaken","orb","peach","pine","plain","pride","proud","rain","rapid","ring","river","rock","rose","rune","silent","silk","silver","sky","snow","spell","spring","spruce","star","stern","stout","sun","swift","thorn","vine","water","weather","willow","wood","yew"];
var nm7 = ["","","","","","","","","","","","","","","bash","beam","belly","bend","berry","bind","blossom","blow","bough","brace","braid","bramble","branch","brand","breath","breeze","brook","brooke","bush","cloud","copse","covert","crest","crown","dance","dancer","dew","down","draft","dream","drift","drop","dust","fall","fern","fir","flow","gaze","gem","glade","gleam","glide","glow","grove","gust","heart","husk","larch","leaf","lock","ridge","river","rock","run","seed","shade","shine","shrub","skipper","snow","soar","song","spell","spirit","sprout","spur","stand","star","stone","stride","stutter","sun","thorn","track","trap","twig","ward","water","wind","wing","wings"];
var nm8 = ["Dancer","Faerie","Flitter","Leprechaun","Pixie","Prankster","Schemer","Spinner","Sprite","Swarm"];

var nm9 = ["Autumn","Blizzard","Brush","Bush","Cloud","Dew","Dewdrop","Dirt","Dream","Dust","Fan","Feather","Fire","Flight","Flock","Flower","Fog","Forest","Frost","Garden","Glen","Gust","Honey","Ice","Icicle","Jewel","Kite","Leaf","Lift","Light","Lunar","Marble","Marsh","Meadow","Mist","Moon","Mountain","Nectar","Night","Nightshade","Ocean","Plane","Rain","Riddle","River","Sand","Sea","Shade","Shadow","Shore","Shrub","Silk","Snow","Solar","Spring","Storm","Stream","Summer","Sun","Thorn","Thunder","Water","Wind","Winter","Zephyr"];
var nm10 = ["Archmage","Blackguard","Caretaker","Clique","Cohort","Dancer","Dewdancer","Disciple","Faerie","Flitter","Gatewarden","Guard","Guardian","Harbinger","Invader","Keeper","Lancer","Leprechaun","Mage","Mechanist","Noble","Overseer","Pixie","Prankster","Priestess","Prowler","Ranger","Rascal","Rover","Sage","Schemer","Scout","Seer","Singer","Skipper","Soulsinger","Spinner","Sprite","Spy","Squadron","Swarm","Thief","Trickster","Wanderer","Warden","Witch"];

var br = "";

function nameGen(){
	$('#placeholder').css('textTransform', 'capitalize');
	var element = document.createElement("div");
	element.setAttribute("id", "result");

	for(i = 0; i < 10; i++){
		if(i < 6){
			rnd = Math.floor(Math.random() * nm1.length);
			rnd2 = Math.floor(Math.random() * nm2.length);
			rnd3 = Math.floor(Math.random() * nm3.length);
			rnd4 = Math.floor(Math.random() * nm2.length);
			rnd5 = Math.floor(Math.random() * nm5.length);
			while(nm3[rnd3] === nm1[rnd]){
				rnd3 = Math.floor(Math.random() * nm3.length);
			}
			if(i < 2){
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm5[rnd5];
			}else{
				rnd6 = Math.floor(Math.random() * nm4.length);
				rnd7 = Math.floor(Math.random() * nm2.length);
				while(nm3[rnd3] === nm4[rnd6]){
					rnd6 = Math.floor(Math.random() * nm4.length);
				}
				if(i < 4){
					names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm4[rnd6] + nm2[rnd7] + nm5[rnd5];
				}else{
					names = nm1[rnd] + nm2[rnd4] + nm4[rnd6] + nm2[rnd2] + nm3[rnd3] + nm2[rnd7] + nm5[rnd5];
				}
			}
		}else if(i < 8){
			rnd = Math.floor(Math.random() * nm6.length);
			rnd2 = Math.floor(Math.random() * nm7.length);
			while(nm6[rnd] === nm7[rnd2]){
				rnd2 = Math.floor(Math.random() * nm7.length);
			}
			rnd3 = Math.floor(Math.random() * nm8.length);
			names = nm6[rnd] + nm7[rnd2] + " " + nm8[rnd3];
		}else{
			rnd = Math.floor(Math.random() * nm9.length);
			rnd2 = Math.floor(Math.random() * nm10.length);
			names = nm9[rnd] + " " + nm10[rnd2];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/mgtFaeries.py
__all__ = ['mgtFaeries']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm8', 'nm3', 'nm7', 'nm9', 'nm10', 'nm2', 'nm6'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if (var.get('i')<Js(6.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                while PyJsStrictEq(var.get('nm3').get(var.get('rnd3')),var.get('nm1').get(var.get('rnd'))):
                    var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                if (var.get('i')<Js(2.0)):
                    var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm5').get(var.get('rnd5'))))
                else:
                    var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                    var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                    while PyJsStrictEq(var.get('nm3').get(var.get('rnd3')),var.get('nm4').get(var.get('rnd6'))):
                        var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                    if (var.get('i')<Js(4.0)):
                        var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd6')))+var.get('nm2').get(var.get('rnd7')))+var.get('nm5').get(var.get('rnd5'))))
                    else:
                        var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd6')))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd7')))+var.get('nm5').get(var.get('rnd5'))))
            else:
                if (var.get('i')<Js(8.0)):
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm6').get('length'))))
                    var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
                    while PyJsStrictEq(var.get('nm6').get(var.get('rnd')),var.get('nm7').get(var.get('rnd2'))):
                        var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
                    var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm8').get('length'))))
                    var.put('names', (((var.get('nm6').get(var.get('rnd'))+var.get('nm7').get(var.get('rnd2')))+Js(' '))+var.get('nm8').get(var.get('rnd3'))))
                else:
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm9').get('length'))))
                    var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm10').get('length'))))
                    var.put('names', ((var.get('nm9').get(var.get('rnd'))+Js(' '))+var.get('nm10').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js(''), Js(''), Js(''), Js(''), Js('c'), Js('f'), Js('h'), Js('l'), Js('n'), Js('r'), Js('s'), Js('w'), Js('z')]))
var.put('nm2', Js([Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('y'), Js('oo'), Js('ia'), Js('ea'), Js('ee')]))
var.put('nm3', Js([Js('b'), Js('d'), Js('l'), Js('ll'), Js('m'), Js('mm'), Js('n'), Js('nn'), Js('r'), Js('s'), Js('v'), Js('b'), Js('d'), Js('dw'), Js('l'), Js('ll'), Js('lw'), Js('lr'), Js('lm'), Js('ln'), Js('m'), Js('mr'), Js('mm'), Js('n'), Js('nm'), Js('nr'), Js('nv'), Js('r'), Js('rl'), Js('rn'), Js('rm'), Js('sh'), Js('sn'), Js('sr'), Js('vl'), Js('vr')]))
var.put('nm4', Js([Js('c'), Js('d'), Js('l'), Js('n'), Js('r'), Js('s'), Js('th'), Js('z')]))
var.put('nm5', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('h'), Js('l'), Js('n'), Js('ph'), Js('s'), Js('th')]))
var.put('nm6', Js([Js('acorn'), Js('alder'), Js('alpen'), Js('amber'), Js('autumn'), Js('barely'), Js('beech'), Js('birch'), Js('briar'), Js('bright'), Js('cedar'), Js('cherry'), Js('cinder'), Js('cloud'), Js('crystal'), Js('dawn'), Js('dew'), Js('dream'), Js('dusk'), Js('elm'), Js('ember'), Js('feather'), Js('fern'), Js('fog'), Js('forest'), Js('free'), Js('frost'), Js('gentle'), Js('grand'), Js('grass'), Js('great'), Js('green'), Js('haven'), Js('hazel'), Js('heart'), Js('holly'), Js('humble'), Js('keen'), Js('kind'), Js('leaf'), Js('light'), Js('lone'), Js('maple'), Js('marble'), Js('marsh'), Js('mellow'), Js('mist'), Js('moon'), Js('moss'), Js('nettle'), Js('night'), Js('oaken'), Js('orb'), Js('peach'), Js('pine'), Js('plain'), Js('pride'), Js('proud'), Js('rain'), Js('rapid'), Js('ring'), Js('river'), Js('rock'), Js('rose'), Js('rune'), Js('silent'), Js('silk'), Js('silver'), Js('sky'), Js('snow'), Js('spell'), Js('spring'), Js('spruce'), Js('star'), Js('stern'), Js('stout'), Js('sun'), Js('swift'), Js('thorn'), Js('vine'), Js('water'), Js('weather'), Js('willow'), Js('wood'), Js('yew')]))
var.put('nm7', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('bash'), Js('beam'), Js('belly'), Js('bend'), Js('berry'), Js('bind'), Js('blossom'), Js('blow'), Js('bough'), Js('brace'), Js('braid'), Js('bramble'), Js('branch'), Js('brand'), Js('breath'), Js('breeze'), Js('brook'), Js('brooke'), Js('bush'), Js('cloud'), Js('copse'), Js('covert'), Js('crest'), Js('crown'), Js('dance'), Js('dancer'), Js('dew'), Js('down'), Js('draft'), Js('dream'), Js('drift'), Js('drop'), Js('dust'), Js('fall'), Js('fern'), Js('fir'), Js('flow'), Js('gaze'), Js('gem'), Js('glade'), Js('gleam'), Js('glide'), Js('glow'), Js('grove'), Js('gust'), Js('heart'), Js('husk'), Js('larch'), Js('leaf'), Js('lock'), Js('ridge'), Js('river'), Js('rock'), Js('run'), Js('seed'), Js('shade'), Js('shine'), Js('shrub'), Js('skipper'), Js('snow'), Js('soar'), Js('song'), Js('spell'), Js('spirit'), Js('sprout'), Js('spur'), Js('stand'), Js('star'), Js('stone'), Js('stride'), Js('stutter'), Js('sun'), Js('thorn'), Js('track'), Js('trap'), Js('twig'), Js('ward'), Js('water'), Js('wind'), Js('wing'), Js('wings')]))
var.put('nm8', Js([Js('Dancer'), Js('Faerie'), Js('Flitter'), Js('Leprechaun'), Js('Pixie'), Js('Prankster'), Js('Schemer'), Js('Spinner'), Js('Sprite'), Js('Swarm')]))
var.put('nm9', Js([Js('Autumn'), Js('Blizzard'), Js('Brush'), Js('Bush'), Js('Cloud'), Js('Dew'), Js('Dewdrop'), Js('Dirt'), Js('Dream'), Js('Dust'), Js('Fan'), Js('Feather'), Js('Fire'), Js('Flight'), Js('Flock'), Js('Flower'), Js('Fog'), Js('Forest'), Js('Frost'), Js('Garden'), Js('Glen'), Js('Gust'), Js('Honey'), Js('Ice'), Js('Icicle'), Js('Jewel'), Js('Kite'), Js('Leaf'), Js('Lift'), Js('Light'), Js('Lunar'), Js('Marble'), Js('Marsh'), Js('Meadow'), Js('Mist'), Js('Moon'), Js('Mountain'), Js('Nectar'), Js('Night'), Js('Nightshade'), Js('Ocean'), Js('Plane'), Js('Rain'), Js('Riddle'), Js('River'), Js('Sand'), Js('Sea'), Js('Shade'), Js('Shadow'), Js('Shore'), Js('Shrub'), Js('Silk'), Js('Snow'), Js('Solar'), Js('Spring'), Js('Storm'), Js('Stream'), Js('Summer'), Js('Sun'), Js('Thorn'), Js('Thunder'), Js('Water'), Js('Wind'), Js('Winter'), Js('Zephyr')]))
var.put('nm10', Js([Js('Archmage'), Js('Blackguard'), Js('Caretaker'), Js('Clique'), Js('Cohort'), Js('Dancer'), Js('Dewdancer'), Js('Disciple'), Js('Faerie'), Js('Flitter'), Js('Gatewarden'), Js('Guard'), Js('Guardian'), Js('Harbinger'), Js('Invader'), Js('Keeper'), Js('Lancer'), Js('Leprechaun'), Js('Mage'), Js('Mechanist'), Js('Noble'), Js('Overseer'), Js('Pixie'), Js('Prankster'), Js('Priestess'), Js('Prowler'), Js('Ranger'), Js('Rascal'), Js('Rover'), Js('Sage'), Js('Schemer'), Js('Scout'), Js('Seer'), Js('Singer'), Js('Skipper'), Js('Soulsinger'), Js('Spinner'), Js('Sprite'), Js('Spy'), Js('Squadron'), Js('Swarm'), Js('Thief'), Js('Trickster'), Js('Wanderer'), Js('Warden'), Js('Witch')]))
pass
pass


# Add lib to the module scope
mgtFaeries = var.to_python()